src/System.Slices/System/Buffers/OwnedMemory.cs

```diff
@@ -178,16 +178,25 @@ bool IMemory<T>.TryGetArray(long id, out ArraySegment<T> buffer)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal Span<T> GetSpanInternal(long id)
+        internal Span<T> GetSpanInternal(long id, int index, int length)
         {
             VerifyId(id);
-            return Span;
+            var array = Array;
+            if (array != null)
+                return new Span<T>(array, Offset + index, length);
+            else unsafe
+                {
+                    if (length > Length - index) throw new ArgumentOutOfRangeException();
+                    var offsetBytes = Unsafe.SizeOf<T>() * index;
```

Comment:
I think there wes problem with multiplication - what happens if the result is more than Int32.MaxInt. Writing correct code that operates with native ints is real pain... .